salt/states/postgres_group.py

```diff
@@ -171,18 +194,25 @@ def present(
             update["replication"] = replication
         if superuser is not None and group_attr["superuser"] != superuser:
             update["superuser"] = superuser
-        if password is not None and (
-            refresh_password or group_attr["password"] != password
-        ):
+        if password is not None:
             update["password"] = True
+        if groups is not None:
+            lgroups = groups
+            if isinstance(groups, ((str,), str)):
```

Comment:
```suggestion
            if isinstance(groups, str):
```

Reply:
Is it possible to get the upgrade script to do this, or is it not smart enough?

Reply:
as you noticed, right now it's not smart enough. It might be possible to fix that with some work.